pkg/server/devfile-handler.go

```diff
@@ -0,0 +1,181 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/console/pkg/serverutils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog"
+
+	devfilev1 "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	devfile "github.com/devfile/library/pkg/devfile"
+	"github.com/devfile/library/pkg/devfile/generator"
+	"github.com/devfile/library/pkg/devfile/parser"
+)
+
+const (
+	dockerfilePath = "Dockerfile"
+)
+
+func (s *Server) devfileHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		data       devfileForm
+		devfileObj parser.DevfileObj
+	)
+
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to decode response: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
```

Comment:
This should probably be 400 bad request instead of internal server error if the request body is wrong.

(Same for the other error responses below.)

Reply:
Good call, I'll update those to `StatusBadRequest`.

Reply:
I think some of these errors are unexpected failures of the library... I think it's probably best to keep those 500 Internal Server Errors

Reply:
50-50 split in the end. Anything that the user can "fix" by modifying their repo I left... the others seem internal failures of the library.